newIDE/app/src/KeyboardShortcuts/ShortcutsList.js

```diff
@@ -0,0 +1,120 @@
+import { Line } from '../UI/Grid';
+import RaisedButton from '../UI/RaisedButton';
+import { type ShortcutMap } from './DefaultShortcuts';
+import { getShortcutDisplayName } from './index';
+import Window from '../Utils/Window';
+import defaultShortcuts from '../KeyboardShortcuts/DefaultShortcuts';
+import ShortcutsListRow from './ShortcutsListRow';
+import commandsList, {
+  type CommandName,
+  commandAreas,
+} from '../CommandPalette/CommandsList';
+
+type Props = {|
+  i18n: I18n,
+  userShortcutMap: ShortcutMap,
+  onEdit: (commandName: CommandName, shortcut: string) => void,
+  onReset: () => void,
+|};
+
+const ShortcutsList = (props: Props) => {
+  const [
+    editedShortcut,
+    setEditedShortcut,
+  ] = React.useState<null | CommandName>(null);
+
+  const resetAllShortcutsToDefault = () => {
+    const answer = Window.showConfirmDialog(
+      props.i18n._(
+        t`Are you sure you want to reset all shortcuts to their default values?`
+      ),
+      'question'
+    );
+    if (answer) props.onReset();
+  };
+
+  const resetShortcut = (commandName: CommandName) => {
+    props.onEdit(commandName, defaultShortcuts[commandName]);
+  };
+
+  const areaWiseCommands = {};
+  Object.keys(commandsList)
+    .filter(name => !commandsList[name].noShortcut)
+    .forEach(name => {
+      const areaName = commandsList[name].area;
+      if (!areaWiseCommands[areaName]) areaWiseCommands[areaName] = [];
+      areaWiseCommands[areaName].push(name);
+    });
+
+  return (
+    <>
+      <Line>
+        <RaisedButton
+          label={<Trans>Reset all shortcuts to default</Trans>}
+          onClick={resetAllShortcutsToDefault}
+        />
+      </Line>
+      <List>
+        {Object.keys(areaWiseCommands).map(areaName => (
+          <React.Fragment key={areaName}>
+            <Text size="title">{props.i18n._(commandAreas[areaName])}</Text>
+            {areaWiseCommands[areaName].map(commandName => {
+              // Get default and user-set shortcuts
+              const userShortcut = props.userShortcutMap[commandName];
+              const defaultShortcut = defaultShortcuts[commandName] || '';
+              const shortcutString = userShortcut || defaultShortcut;
+              const shortcutDisplayName = getShortcutDisplayName(
+                shortcutString
+              );
+              // Check if shortcut clashes with another command
+              const clashingCommandName = Object.keys(commandsList)
+                .filter(commandName => !commandsList[commandName].noShortcut)
+                .find(otherCommandName => {
+                  if (otherCommandName === commandName) return false;
+                  const otherShortcut =
+                    props.userShortcutMap[otherCommandName] ||
+                    defaultShortcuts[otherCommandName] ||
+                    '';
+                  if (shortcutString !== otherShortcut) return false;
+                  if (shortcutString === '') return false;
```

Comment:
In this logic, it seems that the complexity is more important than what it could be. 
You iterate on all commands (well, on all areas, then on each commands of this area, so at the end it's on all commands). For each command, you get its shortcutString (so far so good).
Then inside this loop, you're iterating again on the commandsList, computing for each one its shortcutString.

At the end, this means that you're roughly doing `n^2` the work of computing the shortcut strings and comparing them. When we'll have 200 commands, this means 40000 operations, which start to make a lot for some low end or mobile devices :)

Instead, consider the approach of using an object as a map of used shortcutString. In pseudo code:
```
const shortcutStringToCommands = {};
for each command:
  // Get userShortcut and defaultShortcut as before
  const shortcutString = userShortcut || defaultShortcut
  shortcutStringToCommands[shortcutString] = (shortcutStringToCommands[shortcutString] || []).concat(command);
```

This is doing a single loop on commands, so complexity is `n`. Better, you can put this in a separate function outside of the render.

Then in the render, you iterate on commands again, and for each you look into `shortcutStringToCommands`, with the shortcutString of the command. If the array contains just your command, you're fine. If it contains something else, you have one (or more!) clashing commands :) 

At the end, this is done in roughly `2*n` operations (2 separate loops), so even with 200 commands, we'll have 400 operations to do - it should be more than fast enough even for the future and low end devices :)